service/history/configs/config.go

```diff
@@ -375,8 +378,11 @@ func NewConfig(
 
 		EmitShardLagLog:                       dc.GetBoolProperty(dynamicconfig.EmitShardLagLog, false),
 		HistoryCacheInitialSize:               dc.GetIntProperty(dynamicconfig.HistoryCacheInitialSize, 128),
+		HistoryCacheLimitSizeBased:            dc.GetBoolProperty(dynamicconfig.HistoryCacheSizeBasedLimit, false)(),
```

Comment:
This is what I am doing to avoid HistoryCacheLimitSizeBased changing in runtime. I am calling the GetBoolProperty function here itself and assigning the bool value to HistoryCacheLimitSizeBased.

I considered copying this config to some variable during startup. But workflow.Context is created in many different places.